packages/next/src/builders/build/lib/create-package-json.ts

```diff
@@ -2,23 +2,63 @@ import { BuilderContext } from '@angular-devkit/architect';
 import { writeFileSync } from 'fs';
 import { join } from 'path';
 import { readJsonFile } from '@nrwl/workspace';
+import { createProjectGraph } from '@nrwl/workspace/src/core/project-graph';
+import { calculateProjectDependencies } from '@nrwl/workspace/src/utils/buildable-libs-utils';
 import { NextBuildBuilderOptions } from '../../../utils/types';
 
+function getProjectDeps(context: BuilderContext, rootPackageJson: any) {
+  const projGraph = createProjectGraph();
+  const { dependencies: deps } = calculateProjectDependencies(
+    projGraph,
+    context
+  );
+  const depNames = deps
+    .map((d) => d.node)
+    .filter((node) => node.type === 'npm')
+    .map((node) => node.name.replace('npm:', ''));
+  const dependencies = depNames
+    .filter((packageName) => packageName in rootPackageJson.dependencies)
+    .reduce((deps, pkgName) => {
+      return { ...deps, [pkgName]: rootPackageJson.dependencies[pkgName] };
+    }, {});
+  const devDependencies = depNames
```

Comment:
Do we need the `devDependencies`? Only the dependencies needed at runtime should be added right?

Reply:
Because the `next.config.js` may import packages from the `devDependencies` (such as `@nrwl/next` by default) we need any `devDependencies` imported as well, but it should still only be any found in the `calculateProjectDependencies()` call.